src/ast/nodes/shared/BitFlags.ts

```diff
@@ -0,0 +1,41 @@
+export const enum Flag {
+	included = 0b0000_0000_0000_0000_0000_0000_0000_0001,
+	deoptimized = 0b0000_0000_0000_0000_0000_0000_0000_0010,
+	tdzAccessDefined = 0b0000_0000_0000_0000_0000_0000_0000_0100,
+	tdzAccess = 0b0000_0000_0000_0000_0000_0000_0000_1000,
+	assignmentDeoptimized = 0b0000_0000_0000_0000_0000_0000_0001_0000,
+	bound = 0b0000_0000_0000_0000_0000_0000_0010_0000,
+	isUndefined = 0b0000_0000_0000_0000_0000_0000_0100_0000,
+	optional = 0b0000_0000_0000_0000_0000_0000_1000_0000,
+	async = 0b0000_0000_0000_0000_0000_0001_0000_0000,
+	deoptimizedReturn = 0b0000_0000_0000_0000_0000_0010_0000_0000,
+	computed = 0b0000_0000_0000_0000_0000_0100_0000_0000,
+	hasLostTrack = 0b0000_0000_0000_0000_0000_1000_0000_0000,
+	hasUnknownDeoptimizedInteger = 0b0000_0000_0000_0000_0001_0000_0000_0000,
+	hasUnknownDeoptimizedProperty = 0b0000_0000_0000_0000_0010_0000_0000_0000,
+	directlyIncluded = 0b0000_0000_0000_0000_0100_0000_0000_0000,
+	deoptimizeBody = 0b0000_0000_0000_0000_1000_0000_0000_0000,
+	isBranchResolutionAnalysed = 0b0000_0000_0000_0001_0000_0000_0000_0000,
+	await = 0b0000_0000_0000_0010_0000_0000_0000_0000,
+	method = 0b0000_0000_0000_0100_0000_0000_0000_0000,
+	shorthand = 0b0000_0000_0000_1000_0000_0000_0000_0000,
+	tail = 0b0000_0000_0001_0000_0000_0000_0000_0000,
+	prefix = 0b0000_0000_0010_0000_0000_0000_0000_0000
+}
```

Comment:
Pre-calculated bitmasks are usually faster. 

The naming `flag`, `bitFlag` ... is somewhat unfortunate for bitmasks

Reply:
@jerome-benoit The bit-shift approach would be folded into constants at build time when the typescript is compiled, so they are effectively still pre-calculated. I will make this change as I agree its cleaner. My original source was tab aligned but the code formatter made it harder to read.

As for the naming, perhaps my nomenclature has been corrupted by C# using the `[Flags]` attribute on enums that represent bit-masks. While I agree we could rename `Flags => Mask` and `BitFlags.ts => BitMask.ts`. What would you name the `flags` member variable?

Reply:
> @jerome-benoit The bit-shift approach would be folded into constants at build time when the typescript is compiled, so they are effectively still pre-calculated. I will make this change as I agree it's cleaner. My original source was tab aligned but the code formatter made it harder to read.
> 

Hope that the transpiler used in rollup with not change that behaviour over time. 

> As for the naming, perhaps my nomenclature has been corrupted by C# using the `[Flags]` attribute on enums that represent bit-masks. While I agree we could rename `Flags => Mask` and `BitFlags.ts => BitMask.ts`. What would you name the `flags` member variable?

I think it's more a naming consistency: if the code reference '...flag(s)' everywhere, it should not use '...mask(s)'. I think there's a '...mask(s)' usage in the PR. 
And if the rollup code is using '...mask(s)' elsewhere for bitmasks, I think that PR should align the naming with the rest of the code. If not, using '...flag(s)' should not introduce any namespace inconsistencies.

Reply:
> Hope that the transpiler used in rollup with not change that behaviour over time.

Even if constant folding wasn't applied, they would still only be assigned once at module initialization time when the enum is constructed. Currently it generates like this which is pretty typical and @rollup/plugin-typescript just uses the typescript compiler directly:
```js
var Flag;
(function (Flag) {
    Flag[Flag["included"] = 1] = "included";
    Flag[Flag["deoptimized"] = 2] = "deoptimized";
    Flag[Flag["tdzAccessDefined"] = 4] = "tdzAccessDefined";
    Flag[Flag["tdzAccess"] = 8] = "tdzAccess";
    Flag[Flag["assignmentDeoptimized"] = 16] = "assignmentDeoptimized";
    Flag[Flag["bound"] = 32] = "bound";
    Flag[Flag["isUndefined"] = 64] = "isUndefined";
    Flag[Flag["optional"] = 128] = "optional";
    Flag[Flag["async"] = 256] = "async";
    Flag[Flag["deoptimizedReturn"] = 512] = "deoptimizedReturn";
    Flag[Flag["computed"] = 1024] = "computed";
    Flag[Flag["hasLostTrack"] = 2048] = "hasLostTrack";
    Flag[Flag["hasUnknownDeoptimizedInteger"] = 4096] = "hasUnknownDeoptimizedInteger";
    Flag[Flag["hasUnknownDeoptimizedProperty"] = 8192] = "hasUnknownDeoptimizedProperty";
    Flag[Flag["directlyIncluded"] = 16384] = "directlyIncluded";
    Flag[Flag["deoptimizeBody"] = 32768] = "deoptimizeBody";
    Flag[Flag["isBranchResolutionAnalysed"] = 65536] = "isBranchResolutionAnalysed";
    Flag[Flag["await"] = 131072] = "await";
    Flag[Flag["method"] = 262144] = "method";
    Flag[Flag["shorthand"] = 524288] = "shorthand";
    Flag[Flag["tail"] = 1048576] = "tail";
    Flag[Flag["prefix"] = 2097152] = "prefix";
})(Flag || (Flag = {}));
```

That said given this is a `const enum`, I was hoping it would inline these values in the generated javascript instead of requring a property access, but because the tsconfig.json sets `isolatedModules: true` this cross-file optimization doesn't work. It used to be that rollup-plugin-typescript used the transpileModule to transpile a single file at a time and consequently it was expected that you set `isolatedModules: true` to avoid issues relating to how its compiled.

Alternatively, we could just export individual constants `export FLAG_included = 1` to work around this, but it is kind of a pain to need to import each flag you are using individually. There are no other changes to the generated output when I diff before/after removing isolatedModules from the tsconfig, so I'm leaning toward doing that.

Reply:
I checked and yes, `isolatedModules` was a relic from the time we used the old `rollup-plugin-typescript`. Please remove :)

Reply:
> but it is kind of a pain to need to import each flag you are using individually.

If we find we need to switch back to isolatedModules in the future, we could alleviate this pain by putting all exports in a single file that contains nothing but these constants and then use a `*` import, e.g. `import * as flags from '...`, and then you can write `flags.FLAG_INCLUDED` at the usage site while Rollup would directly replace this with `FLAG_INCLUDED`.